partner_replenishment_block_archive/models/res_partner.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Akretion (https://www.akretion.com).
+# @author Mathieu DELVA <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def write(self, vals):
+        if "active" in vals and not vals["active"]:
+            replenishements = self.env["stock.warehouse.orderpoint"].search(
+                [("supplier_id", "=", self.id)]
```

Comment:
Won't work when updating many records.
You can also make it more efficient.
```suggestion
                [("supplier_id", "in", self.ids)], limit=1
```